### PR TITLE
Add CPB test

### PR DESCRIPTION
**Pull request by pkubaj**
keys.robot is included in the pcengines.robot file because keys are actually used there.

### CODE REVIEW

**Comment by pkubaj** [comment]
logs:
[logs.tar.gz](https://github.com/Dasharo/open-source-firmware-validation/files/14607523/logs.tar.gz)

File: dasharo-performance/core-performance-boost.robot [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Resource            ../variables.robot
+Resource            ../keywords.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+CPB001.001 Check if disabling CPB decrease performance
+    Skip If    not ${CPB_SUPPORT}    CBP not supported
```
Comment (macpijan) [inline dasharo-performance/core-performance-boost.robot:15]: These options are all part of APU menu, so maybe we ca reuse the single flag as here: https://github.com/Dasharo/open-source-firmware-validation/pull/221/files#diff-673a70ee5ebe88c297daaa3c52dc4d6e47b8440b7ab1b02489daaed919a15f20R33 ?

Maybe these should even be integrated into a single test suite.

File: dasharo-performance/core-performance-boost.robot [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${apu_menu}=    Enter Dasharo Submenu    ${setup_menu}    Dasharo APU Configuration
+    Set Option State    ${apu_menu}    Core Performance Boost    ${FALSE}
+    Save Changes And Reset
+    Boot System Or From Connected Disk    debian
+    Login To Linux
+    ${first_check}=    Execute Linux Command
```
Comment (macpijan) [inline dasharo-performance/core-performance-boost.robot:23]: Please try to use terminal keywords as suggested here: https://github.com/Dasharo/open-source-firmware-validation?tab=readme-ov-file#keywords
We have some redundancy in keywords and do not want to make it even more.

File: dasharo-performance/core-performance-boost.robot (hunk 1 above)
Comment (mkopec) [reply dasharo-performance/core-performance-boost.robot:15]: Yeah that's what I had in mind when I created [that file](https://github.com/Dasharo/open-source-firmware-validation/pull/221/files#diff-673a70ee5ebe88c297daaa3c52dc4d6e47b8440b7ab1b02489daaed919a15f20R33) , that all of the APU specific tests would go there.

Comment (pkubaj) [reply dasharo-performance/core-performance-boost.robot]: Done.

File: dasharo-performance/core-performance-boost.robot (hunk 2 above)
Comment (pkubaj) [reply dasharo-performance/core-performance-boost.robot]: Done.

**Comment by macpijan** [comment]
@pkubaj other pce-changes already in develop branch, please rebase and change target

**Comment by pkubaj** [comment]
@macpijan  Please check, files changed seem ok.

File: keywords.robot [hunk 3]
```diff
@@ -74,6 +74,9 @@ Login To Windows
     IF    '${DUT_CONNECTION_METHOD}' == 'Telnet'
         Set Test Variable    ${DUT_CONNECTION_METHOD}    SSH
     END
+    IF    '${DUT_CONNECTION_METHOD}' == 'Telnet'
+        Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    END
```
Comment (miczyg1) [inline keywords.robot:79]: Please put that in a single IF:

```
    IF    '${DUT_CONNECTION_METHOD}' == 'Telnet'
        Set Test Variable    ${DUT_CONNECTION_METHOD}    SSH
        Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
    END
```

Also if we do `Set Global Variable`, is `Set Test Variable` still needed?

Comment (pkubaj) [reply keywords.robot:79]: I'm not sure, those are actually not my changes.

Comment (pkubaj) [reply keywords.robot]: Done per suggestion.

Comment (pkubaj) [reply keywords.robot]: Looking at this fragment of code, it checks whether connection method is telnet and if it is, it switches connection method to SSH. The next conditional checks whether connection method is SSH. So we could just do something like:
```
    IF    '${DUT_CONNECTION_METHOD}' == 'SSH' or '${DUT_CONNECTION_METHOD}' == 'Telnet'
        Login To Windows Via SSH    ${DEVICE_WINDOWS_USERNAME}    ${DEVICE_WINDOWS_PASSWORD}
    ELSE
        Fail    Login to Windows not supported. DUT_CONNECTION_METHOD must be set to SSH.
    END
```

**Comment by mkopec** [comment]
Shouldn't this PR be targeting `configs-rework` instead of `develop`? I'm seeing a lot of unrelated changes here

**Comment by mkopec** [comment]
I'm seeing a lot of unrelated changes here. Could you rebase your changes on top of the latest `develop` so that only your changes are in this PR?

File: keywords.robot (hunk 3 above)
Comment (macpijan) [reply keywords.robot]: But these are in your commit: 14f28b7312c7a7b4f26c8530382f93a7f94125ff

I see you have merged develop into your branch, a better idea is to use rebase when updating your branch. We are in favor of avoiding merge commits this way.

**Comment by macpijan** [comment]
This PR is replaced with: https://github.com/Dasharo/open-source-firmware-validation/pull/235

@pkubaj please have a look, I have cleaned it up, including commit messag format we would prefer to keep.

**Pull request closed without merging**